aws/resource_aws_elasticsearch_domain.go

```diff
@@ -230,6 +251,20 @@ func resourceAwsElasticSearchDomainCreate(d *schema.ResourceData, meta interface
 		}
 	}
 
+	if v, ok := d.GetOk("vpc_options"); ok {
+		options := v.([]interface{})
+
+		if len(options) > 1 {
+			return fmt.Errorf("Only a single vpc_options block is expected")
+		} else if len(options) == 1 {
```

Comment:
The else is not needed since the first if returns, meaning we can simplify this to:

```go
if len(options) > 1 {
    return fmt.Errorf("Only a single vpc_options block is expected")
}

if len(options) == 1 {
...
}
```

However, I do think we can simplify this a lot. 
Since only a single `vpc_options` section should be configurable as far as I understand the API, checking more than 2 is unnecessary, hence the `MaxItems` option.

Then, we can also add this line 256:

```go
option, ok := options[0].(map[string]interface{})

if !ok {
    return errors.New("vpc_options is <nil>")
}

if option != nil {
    //...
}
```

The same kind of logic applied to the update section would then be required. What do you think? 😄